src/v2/components/ecs-service/index.ts

```diff
@@ -190,29 +188,34 @@ export class EcsService extends pulumi.ComponentResource {
     super('studion:ecs:Service', name, {}, opts);
     const argsWithDefaults = Object.assign({}, defaults, args);
     const taskExecutionRoleInlinePolicies = pulumi.output(
-      args.taskExecutionRoleInlinePolicies || defaults.taskExecutionRoleInlinePolicies
+      args.taskExecutionRoleInlinePolicies ||
+        defaults.taskExecutionRoleInlinePolicies,
     );
     const taskRoleInlinePolicies = pulumi.output(
-      args.taskRoleInlinePolicies || defaults.taskRoleInlinePolicies
+      args.taskRoleInlinePolicies || defaults.taskRoleInlinePolicies,
     );
     this.name = name;
     this.securityGroups = [];
     this.vpc = pulumi.output(argsWithDefaults.vpc);
     this.logGroup = this.createLogGroup();
-    this.taskExecutionRole = this.createTaskExecutionRole(taskExecutionRoleInlinePolicies);
-    this.taskRole = this.createTaskRole(taskRoleInlinePolicies)
+    this.taskExecutionRole = this.createTaskExecutionRole(
+      taskExecutionRoleInlinePolicies,
+    );
+    this.taskRole = this.createTaskRole(taskRoleInlinePolicies);
 
-    if (argsWithDefaults.volumes.length) {
-      this.persistentStorage = this.createPersistentStorage(this.vpc);
-    }
+    pulumi.output(argsWithDefaults.volumes).apply(volume => {
```

Comment:
Should we unwrap the value within the method to keep the signature of the calls in the constructor on the same abstraction level (with signatures using Pulumi Input -> Output)?